src/main/java/slimeknights/tconstruct/smeltery/recipe/ICastingInventory.java

```diff
@@ -0,0 +1,17 @@
+package slimeknights.tconstruct.smeltery.recipe;
+
+import net.minecraft.fluid.Fluid;
+import slimeknights.tconstruct.library.recipe.inventory.ISingleItemInventory;
+
+public interface ICastingInventory extends ISingleItemInventory {
+  /**
+   * Gets the contained fluid in this inventory
+   * @return  Contained fluid
+   */
+  Fluid getFluid();
+
+  /**
+   * Sets the contained fluid in this inventory
+   */
+  void setFluid(Fluid fluid);
```

Comment:
Don't put `setFluid` here, the interface is intended to be read only, recipes never need to change it (and it will lead to super buggy behavior if they do change it). Its fine to leave in the wrapper.

Plus, based on the specific implementation we may not have a settable fluid. For instance, someone makes an infinite water caster.